simulators/optimism/l1ops/deposit_tests.go

```diff
@@ -299,6 +299,12 @@ func failingDepositWithMintTest(t *hivesim.T, env *optimism.TestEnv) {
 	revertTx, err := failureContract.Fail(l2Opts)
 	require.NoError(t, err)
 
+	// Create transaction that burns more gas on L2 than allotted
+	// in the deposit transaction.
+	l2Opts.GasLimit = 3_000_000
+	burnTx, err := failureContract.FailWithBurn(l2Opts, big.NewInt(2_000_000))
```

Comment:
At 2M burned, and 3M provided, isn't it supposed to pass without exceeding the tx gas limit?

Reply:
Doesn't the gas limit get pulled from the `DepositTransaction` call below?

Reply:
```
tx, err := depositContract.DepositTransaction(
			opts,
			depositor,
			mintAmount,
			1_000_000,
			false,
			data,
		)
```

Reply:
Oh I see, you're preparing the tx here, but then copying the data and repeating it with a lower limit, nice.